src/interpolation/details/ArborX_InterpDetailsCompactRadialBasisFunction.hpp

```diff
@@ -0,0 +1,94 @@
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+#define ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX::Interpolation
+{
+
+#define CRBF_DECL(NAME)                                                        \
+  namespace Details                                                            \
+  {                                                                            \
+  template <std::size_t>                                                       \
+  struct NAME;                                                                 \
+  }                                                                            \
+                                                                               \
+  namespace CRBF                                                               \
+  {                                                                            \
+  template <std::size_t I>                                                     \
+  static constexpr Details::NAME<I> NAME{};                                    \
+  }
+
+#define CRBF_DEF(NAME, N, FUNC)                                                \
+  namespace Details                                                            \
+  {                                                                            \
+  template <>                                                                  \
+  struct NAME<N>                                                               \
+  {                                                                            \
+    template <typename T>                                                      \
+    KOKKOS_INLINE_FUNCTION static constexpr T apply(T const y)                 \
+    {                                                                          \
+      T const x = Kokkos::min(Kokkos::abs(y), T(1));                           \
+      return Kokkos::abs(FUNC);                                                \
+    }                                                                          \
+                                                                               \
+    template <typename T>                                                      \
+    KOKKOS_INLINE_FUNCTION constexpr T operator()(T const y) const             \
+    {                                                                          \
+      return NAME<N>::apply(y);                                                \
+    }                                                                          \
```

Comment:
Why do we need both? Maybe, just
```suggestion
    template <typename T>                                                      \
    KOKKOS_INLINE_FUNCTION static constexpr T evaluate(T const y)                 \
    {                                                                          \
      T const x = Kokkos::min(Kokkos::abs(y), T(1));                           \
      return Kokkos::abs(FUNC);                                                \
    }
```

Reply:
I wanted to be able to do `CRBF::Wendland<2>(value)`, but `operator()` is used nowhere.